File: 2 semestre/Técnica de Desenvolvimento de Algoritmos/Listas de Exercicios/Lista 4/ex14.c
#include <stdio.h>

main()
{
    printf("Compactando um vetor\n\n");

    //vars
    int v[5], i, j;

    //Entrada
    for( i=0; i<5; i++)
    {
        printf("Digite v[%d]: ", i);
        scanf("%d", &v[i]);
    }

    //Processamento
    for(i=4; i>0; i--)
    {
        if(v[i-1] == 0)
        {
            for(j=i-1; j<4; j++)
            {
                v[j] = v[j+1];
                v[j+1] = 0;
            }
        }
    }

    //Saída
    printf("\nVetor Resultante: \n");
    for( i=0; i<5; i++)
    {
        printf("%d\n", v[i]);
    }
}

File: 2 semestre/Laboratório de Desenvolvimento de Algoritmos/Lista de Exercícios 4/src/Ex14.java
//Importando Scanner
import java.util.Scanner;

public class Ex14 {

	public static void main(String[] args){
		
		//Instanciando Scanner
		Scanner input = new Scanner(System.in);
		
		System.out.println("Zeros no Final do Vetor\n\n");
		
		//vars
		int v[] = new int[5];
		int i, j, aux;
		
		//Entrada
		for( i=0; i<5; i++)
		{
			System.out.print("Digite v[" + i + "]:");
			v[i] = input.nextInt();
		}
		
		//Processamento
		for(i=0; i<5; i++)
		{
			if( v[i] == 0)
			{
				for(j=i; j<4; j++)
				{
					aux = v[j];
					v[j] = v[j+1];
					v[j+1] = aux;
				}
			}
		}
			
			//Saída
			System.out.println("\n\nVetor Resultante: ");
			for( i=0; i<5; i++)
			{
				System.out.printf("%d  ", v[i]);
			}
		
		
	}
}
